Sources/XcodeGenKit/PBXProjGenerator.swift

```diff
@@ -219,8 +242,72 @@ public class PBXProjGenerator {
             pbxProject.projects = subprojects
         }
 
-        try project.targets.forEach(generateTarget)
-        try project.aggregateTargets.forEach(generateAggregateTarget)
+        let generateTargetsGroup = DispatchGroup()
+        var targetErrors: [Error] = []
+        var aggregateTargetsErrors: [Error] = []
+
+        var generatedTargetNames: [String] = []
+        var targetsPendingDependencyGeneration: [Target] = project.targets
+        while generatedTargetNames.count != project.targets.count {
```

Comment:
This while loops helps us parallelize more and more targets in each loop, starting with targets, which have no dependencies at all